spec/openapi.yaml

```
@@ -94,10 +94,98 @@ security:
+        - $ref: "#/components/parameters/format"
+        - $ref: "#/components/parameters/limit"
+        - $ref: "#/components/parameters/offset"
+        - $ref: "#/components/parameters/created_start_date"
+        - $ref: "#/components/parameters/created_end_date"
+        - $ref: "#/components/parameters/changed_start_date"
+        - $ref: "#/components/parameters/changed_end_date"
+        - $ref: "#/components/parameters/ids"
+        - $ref: "#/components/parameters/uuids"
+      responses:
+        "200":
+          description: Product list
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Article"
+            text/xml:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Article"
+  "/v{version}/products/{product_uuid}":
+    get:
+      tags:
+        - Products
+      summary: Return a single product
+      description: Returns a specific product
+      operationId: getProduct
+      parameters:
+        - $ref: "#/components/parameters/version"
+        - $ref: "#/components/parameters/language"
+        - $ref: "#/components/parameters/format"
+        - $ref: "#/components/parameters/use_url_id"
+        - name: product_uuid
+          in: path
+          description: Product UUID
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        "200":
+          description: Product details
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Article"
+            text/xml:
+              schema:
+                $ref: "#/components/schemas/Article"
+  "/v{version}/products/count":
+    get:
+      tags:
+        - Products
+      summary: Return the number of available products
+      description: Returns the number of available products
+      operationId: getProductCount
+      parameters:
+        - $ref: "#/components/parameters/version"
+        - $ref: "#/components/parameters/language"
+        - $ref: "#/components/parameters/format"
+        - $ref: "#/components/parameters/created_start_date"
+        - $ref: "#/components/parameters/created_end_date"
+        - $ref: "#/components/parameters/changed_start_date"
+        - $ref: "#/components/parameters/changed_end_date"
+      responses:
+        "200":
+          description: Number of available articles
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/CountResponse"
+            text/xml:
+              schema:
+                $ref: "#/components/schemas/CountResponse"
   "/v{version}/articles":
     get:
       tags:
         - Articles
+      deprecated: true
       summary: Return a collection of articles
       description: Returns a listing of articles for the store
```

would add a reason why this is deprecated:
```suggestion
      description: Returns a listing of articles for the store. This operation is deprecated and will be removed in the future use getProducts
```